### PR TITLE
[MERGE] event_*: refactor and improve badge reporting

### PR DESCRIPTION
**Pull request by awa-odoo**
PURPOSE

Rework the existing badge report to ease its override and make it look better
and introduce a new ticket report that contains more detailed information.

SPECS

Improve the existing badge report
Change the event badge report to create a proper action report so that it
shows accordingly on form (and other) views.
Remove the duplicate template for event badges and unify into one template
The "event badges" report can be printed from 2 different sources, from the
event.event itself as a preview and from the event.registration as an actual
badge for a specific attendee.
Improve the general layout of the report and remove unnecessary fields.
Introduce a new badge report with the A4 format.

This report is meant to be printed for the attendee and contains more
detailed information that the existing "badge report", which is only meant to
be folded and used as a badge.

We also introduce a new field that allows easily customizing these reports by
adding "extra instructions".

See underlying commits for more details.

Task-2120259

### CODE REVIEW

File: addons/website_event/models/event_event.py [hunk 1]
```diff
@@ -357,15 +357,6 @@ def _track_subtype(self, init_values):
             return self.env.ref('website_event.mt_event_unpublished')
         return super(Event, self)._track_subtype(init_values)
 
-    def action_open_badge_editor(self):
```
Comment (tde-banana-odoo) [inline addons/website_event/models/event_event.py:360]: No more badge editor action ? (reading commit 1)

File: addons/event/report/event_event_reports.xml [hunk 2]
```diff
@@ -16,6 +16,45 @@
...
+        <field name="dpi">96</field>
+    </record>
+
+    <record id="action_report_event_registration_full_page_ticket" model="ir.actions.report">
+        <field name="name">Full Page Ticket</field>
+        <field name="model">event.registration</field>
+        <field name="report_type">qweb-pdf</field>
+        <field name="report_name">event.event_registration_report_template_full_page_ticket</field>
+        <field name="report_file">event.event_registration_report_template_full_page_ticket</field>
+        <field name="print_report_name">'Full Page Ticket - %s - %s' % ((object.event_id.name or 'Event').replace('/',''), object.name.replace('/',''))</field>
```
Comment (tde-banana-odoo) [inline addons/event/report/event_event_reports.xml:40]: Somehow we should probably introduce a "get_printable_name" method someday :D

File: addons/event/data/event_demo.xml [hunk 3]
```diff
@@ -14,6 +14,14 @@
         <field name="event_type_id" ref="event_type_0"/>
         <field name="stage_id" ref="event_stage_booked"/>
         <field name="tag_ids" eval="[(4, ref('event.event_tag_category_1_tag_1')), (4, ref('event.event_tag_category_2_tag_1'))]"/>
+        <field name="ticket_instructions" type="html">
```
Comment (tde-banana-odoo) [inline addons/event/data/event_demo.xml:17]: Would add it to our famous event online demo event also :D

File: addons/event/report/event_event_reports.xml [hunk 4]
```diff
@@ -1,41 +1,82 @@
...
+        <field name="page_width">0</field>
+        <field name="orientation">Portrait</field>
+        <field name="margin_top">0</field>
+        <field name="margin_bottom">8</field>
+        <field name="margin_left">0</field>
+        <field name="margin_right">0</field>
+        <field name="dpi">96</field>
+    </record>
+
+    <record id="action_report_event_registration_full_page_ticket" model="ir.actions.report">
```
Comment (tde-banana-odoo) [inline addons/event/report/event_event_reports.xml:34]: Note that apart that 'it contains more information', it is not really clear what are the use of those two reports ... Could be interesting to explain what they contain, what they do not contain and in which case they should be used.

File: addons/event/report/event_event_templates.xml [hunk 5]
```diff
@@ -1,104 +1,241 @@
...
-                                        <div t-field="event.event_logo">
-                                            <img t-att-src="image_data_uri(event.organizer_id.company_id.logo_web)" style="max-height:1cm; max-width:4cm;" alt="Logo"/>
-                                        </div>
-                                    </span>
-                                    <h4 t-field="event.name"/>
-                                    <h5>( <i class="fa fa-clock-o" aria-label="Period" role="img" title="Period"></i> <span itemprop="startDate" t-field="event.date_begin" t-options='{"hide_seconds": True}'> </span> <i>to</i> <span itemprop="endDate" t-field="event.date_end" t-options='{"hide_seconds": True}'> </span> )</h5>
+<!-- EVENT FULL PAGE TICKET -->
+
+<template id="event_report_template_full_page_ticket">
+    <div class="row page">
```
Comment (tde-banana-odoo) [inline addons/event/report/event_event_templates.xml:107]: Such complicated dom ... will not say "clean your dom", but hey can't we try to remove some nested divs ?

File: addons/event/report/event_event_templates.xml [hunk 6]
```diff
@@ -1,104 +1,241 @@
...
+                                    <div class="o_event_full_page_ticket_font_faded">
+                                        Event Ticket For
+                                    </div>
+                                    <h5 class="o_event_full_page_ticket_event_name font-weight-bold py-0 my-0" t-field="event.name"/>
+                                    <span itemprop="startDate" t-esc="event.date_begin.date()"
+                                        class="font-weight-bold"/>
+                                    <span itemprop="startDateTime" t-esc="event.date_begin" class="font-weight-bold"
+                                        t-options='{"widget": "datetime", "time_only": True, "hide_seconds": True}'/>
+                                    <span class="fa fa-arrow-right o_event_full_page_ticket_font_faded"/>
+                                    <span t-if="event.date_begin.date() != event.date_end.date()"
```
Comment (tde-banana-odoo) [inline addons/event/report/event_event_templates.xml:124]: Can't we use is_one_day field ?

File: addons/event/report/event_event_templates.xml [hunk 7]
```diff
@@ -1,104 +1,241 @@
...
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<template id="event_registration_report_template_foldable_badge">
+    <t t-call="web.basic_layout">
+        <t t-foreach="docs" t-as="attendee">
+            <t t-set="event" t-value="attendee.event_id.with_context(tz=attendee.event_id.date_tz)"/>
```
Comment (tde-banana-odoo) [inline addons/event/report/event_event_templates.xml:89]: Can't we use _set_tz_context method ?

File: addons/event/report/event_event_templates.xml [hunk 8]
```diff
@@ -1,104 +1,241 @@
...
+                <div class="o_event_foldable_badge_ticket_wrapper p-2">
+                    <div class="o_event_foldable_badge_ticket_wrapper_top page">
+                        <h5 class="o_event_foldable_badge_event_name font-weight-bold text-center" t-field="event.name"/>
+                        <div class="text-center o_event_foldable_badge_font_small">
+                            <span itemprop="startDate" t-esc="event.date_begin.date()"
+                                class="font-weight-bold"/>
+                            <span itemprop="startDateTime" t-esc="event.date_begin" class="font-weight-bold"
+                                t-options='{"widget": "datetime", "time_only": True, "hide_seconds": True}'/>
+                            <span class="fa fa-arrow-right o_event_foldable_badge_font_faded"/>
+                            <span t-if="event.date_begin.date() != event.date_end.date()"
```
Comment (tde-banana-odoo) [inline addons/event/report/event_event_templates.xml:25]: Same here, would use is_one_day

File: addons/event/report/event_event_templates.xml [hunk 9]
```diff
@@ -1,104 +1,241 @@
...
+        <main>
+            <t t-out="0"/>
+        </main>
+    </div>
+
+    <div class="row footer o_event_full_page_ticket_footer d-block">
+        <div class="o_event_full_page_ticket_powered_by text-white text-center p-2 w-100">
+            <t t-if="event.organizer_id">
+                <span class="font-weight-bold" t-field="event.organizer_id.name"></span>
+                <span t-if="event.organizer_id.phone" class="pl-3 fa fa-phone"/>
```
Comment (tde-banana-odoo) [inline addons/event/report/event_event_templates.xml:179]: Somehow could be great to use phone_sanitized, but I think this should be done in event_sms (aka: well tant pis probably).

File: addons/event/report/event_event_templates.xml [hunk 10]
```diff
@@ -1,104 +1,241 @@
...
+                <span t-if="event.organizer_id.phone" t-field="event.organizer_id.phone"/>
+                <span t-if="event.organizer_id.email" class="pl-3 fa fa-envelope"/>
+                <span t-if="event.organizer_id.email" t-field="event.organizer_id.email"/>
```
Comment (tde-banana-odoo) [inline addons/event/report/event_event_templates.xml:182]: email_normalized maybe ? that way we have a standard display containing only email and not `"Raoul Poilvache" <blah@bla.blah>`

File: addons/event_sale/report/event_event_templates.xml [hunk 11]
```diff
@@ -1,24 +1,22 @@
...
-                <div style="background: lightgrey; height: 65px;" class="mt16 text-center">
-                    <h3>Ticket Type</h3>
-                </div>
+        <xpath expr="//div[hasclass('o_event_full_page_ticket_side_info')]" position="inside">
+            <div t-if="attendee and attendee.sale_order_id" class="mb-2">
+                <div class="o_event_full_page_ticket_font_faded o_event_full_page_ticket_small_caps font-weight-bold">Order Date</div>
+                <div class="o_event_full_page_ticket_small" t-esc="attendee.sale_order_id.date_order.date()"/>
+            </div>
+            <div t-if="attendee and attendee.event_ticket_id.price">
+                <div class="o_event_full_page_ticket_font_faded o_event_full_page_ticket_small_caps font-weight-bold">Price</div>
```
Comment (tde-banana-odoo) [inline addons/event_sale/report/event_event_templates.xml:16]: Not sure, but shoudln't we use what is defined on so line as prive may have been updated in so itself ? or ballek ?

File: addons/website_event_exhibitor/report/website_event_exhibitor_reports.xml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="event.paperformat_event_full_page_ticket" model="report.paperformat">
+        <!-- Increase bottom margin to leave room to display the sponsor images -->
+        <field name="margin_bottom">29</field>
```
Comment (tde-banana-odoo) [inline addons/website_event_exhibitor/report/website_event_exhibitor_reports.xml:6]: 28 !

**Review by tde-banana-odoo — COMMENTED** [review]
Review !

**Review by awa-odoo — COMMENTED** [review]
Hello @tde-banana-odoo 

It took me some time, but I'm pretty sure I addressed everything.

Thanks for the riviou!

File: addons/event/data/event_demo.xml (hunk 3 above)
Comment (tde-banana-odoo) [reply addons/event/data/event_demo.xml:17]: Nouredine had to remove the "online" word from "online event". We could re-try to do it and add ticket instructions :D #perfectdemodata (ok, don't do it)

File: addons/event/models/event_event.py [hunk 13]
```diff
@@ -434,6 +435,12 @@ def _compute_note(self):
             if event.event_type_id and not is_html_empty(event.event_type_id.note):
                 event.note = event.event_type_id.note
 
+    @api.depends('event_type_id')
+    def _compute_ticket_instructions(self):
+        for event in self:
+            if not event.ticket_instructions and event.event_type_id.ticket_instructions:
```
Comment (tde-banana-odoo) [inline addons/event/models/event_event.py:441]: Shoudln't we use is html empty here ?

File: addons/event/models/event_registration.py [hunk 14]
```diff
@@ -155,7 +155,8 @@ def create(self, vals_list):
         # some were created already open
         if registrations._check_auto_confirmation():
             registrations.sudo().action_confirm()
-        else:
+        elif not self.env.context.get('install_mode', False):
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/event/models/event_registration.py:158]: Would add a small comment in code then.

File: addons/event/models/event_registration.py [hunk 15]
```diff
@@ -167,7 +168,8 @@ def write(self, vals):
 
         ret = super(EventRegistration, self).write(vals)
 
-        if vals.get('state') == 'open':
+        if vals.get('state') == 'open' and not self.env.context.get('install_mode', False):
+            # avoid running scheduler for demo data
```
Comment (tde-banana-odoo) [inline addons/event/models/event_registration.py:172]: Maybe tell why we don't want to run scheduler, not what we are actually doing :)

File: addons/event/report/event_event_reports.xml (hunk 4 above)
Comment (tde-banana-odoo) [reply addons/event/report/event_event_reports.xml:41]: Cool !

File: addons/event/report/event_event_templates.xml (hunk 5 above)
Comment (tde-banana-odoo) [reply addons/event/report/event_event_templates.xml:107]: :it:

File: addons/event/views/event_views.xml [hunk 16]
```diff
@@ -274,7 +279,12 @@
                                 </field>
                             </page>
                             <page string="Notes" name="event_notes">
-                                <field name="note" placeholder="Add a note..."/>
+                                <group>
+                                    <label for="note" string="Note"/><br/>
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/event/views/event_views.xml:229]: Mmmh indeed, html fields on next line are not managed easily through standard group / form view magic. Ok then, some br will not kill people compared to what is already done elsewhere ...

**Review by tde-banana-odoo — APPROVED** [review]

**Review by awa-odoo — COMMENTED** [review]
@tde-banana-odoo should be all good sir!

Thanks for the follow-up.

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**